test/extensions/filters/http/wasm/test_data/metadata_cpp.cc

```diff
@@ -34,7 +34,7 @@ FilterHeadersStatus ExampleContext::onRequestHeaders() {
   if (r != WasmResult::Ok) {
     logDebug(toString(r));
   }
-  r = setMetadataStringValue(MetadataType::Request, "wasm_request_set_key", "wasm_request_set_value");
+  r = setStateStringValue("wasm_request_set_key", "wasm_request_set_value");
```

Comment:
nit: setStateStringValue is a difficult name to parse. What is the lifetime of this "set". Is it state of the request or the filter?